src/components/PopupWithForm.js

```diff
@@ -21,9 +23,11 @@ export default class PopupWithForm extends Popup {
     return this._popupForm;
   }
 
-  // close popup
+  // close popup and reset form along with button text
   close() {
     super.close();
+    this._popupForm.reset();
+    this._renderLoading(false); // reset button when closed
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.